File: src/main/java/no/novari/flyt/configuration/validation/valueparsability/validators/UrlParsabilityValidator.java
package no.novari.flyt.configuration.validation.valueparsability.validators;

import no.novari.flyt.configuration.model.configuration.entities.ValueMapping;
import no.novari.flyt.configuration.validation.valueparsability.ValueParsabilityValidator;
import org.springframework.stereotype.Service;

import java.net.URI;
import java.net.URL;

@Service
public class UrlParsabilityValidator implements ValueParsabilityValidator {

    @Override
    public ValueMapping.Type getTypeToValidate() {
        return ValueMapping.Type.URL;
    }

    @Override
    public boolean isValid(String value) {
        try {
            URI uri = URI.create(value);
            URL ignored = uri.toURL();
            return true;
        } catch (Exception e) {
            return false;
        }
    }

}

File: src/main/kotlin/no/novari/flyt/configuration/validation/valueparsability/validators/UrlParsabilityValidator.kt
package no.novari.flyt.configuration.validation.valueparsability.validators

import no.novari.flyt.configuration.model.configuration.entities.ValueMapping
import no.novari.flyt.configuration.validation.valueparsability.ValueParsabilityValidator
import org.springframework.stereotype.Service
import java.net.URI

@Service
class UrlParsabilityValidator : ValueParsabilityValidator {
    override fun getTypeToValidate(): ValueMapping.Type = ValueMapping.Type.URL

    override fun isValid(value: String): Boolean =
        try {
            URI.create(value).toURL()
            true
        } catch (_: Exception) {
            false
        }
}
